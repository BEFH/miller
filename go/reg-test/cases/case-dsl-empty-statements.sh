run_mlr -n put -v ''
mlr_expect_fail -n filter -v ''

run_mlr -n put -v 'begin {}'
run_mlr -n put -v 'begin {;}'
run_mlr -n put -v 'begin {;;}'
run_mlr -n put -v 'begin {;;;}'
run_mlr -n put -v 'begin {@x=1}'
run_mlr -n put -v 'begin {@x=1;}'
run_mlr -n put -v 'begin {;@x=1}'
run_mlr -n put -v 'begin {@x=1;@y=2}'
run_mlr -n put -v 'begin {@x=1;;@y=2}'

run_mlr -n put -v 'true {}'
run_mlr -n put -v 'true {;}'
run_mlr -n put -v 'true {;;}'
run_mlr -n put -v 'true {;;;}'
run_mlr -n put -v 'true {@x=1}'
run_mlr -n put -v 'true {@x=1;}'
run_mlr -n put -v 'true {;@x=1}'
run_mlr -n put -v 'true {@x=1;@y=2}'
run_mlr -n put -v 'true {@x=1;;@y=2}'

run_mlr -n put -v 'end {}'
run_mlr -n put -v 'end {;}'
run_mlr -n put -v 'end {;;}'
run_mlr -n put -v 'end {;;;}'
run_mlr -n put -v 'end {@x=1}'
run_mlr -n put -v 'end {@x=1;}'
run_mlr -n put -v 'end {;@x=1}'
run_mlr -n put -v 'end {@x=1;@y=2}'
run_mlr -n put -v 'end {@x=1;;@y=2}'
