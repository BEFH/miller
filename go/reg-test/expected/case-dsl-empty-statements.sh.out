
mlr -n put -v 
RAW AST:

AST ROOT:
(null)

BLOCKED AST:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n filter -v 
RAW AST:

AST ROOT:
(null)
mlr: filter statement must not be empty.

mlr -n put -v begin {}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="begin", type=BEGIN:
        text="begin_block", type=STATEMENT_BLOCK:

BLOCKED AST:

BEGIN-BLOCK:
text="begin", type=BEGIN: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="begin_block", type=STATEMENT_BLOCK:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v begin {;}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="begin", type=BEGIN:
        text="begin_block", type=STATEMENT_BLOCK:

BLOCKED AST:

BEGIN-BLOCK:
text="begin", type=BEGIN: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="begin_block", type=STATEMENT_BLOCK:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v begin {;;}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="begin", type=BEGIN:
        text="begin_block", type=STATEMENT_BLOCK:

BLOCKED AST:

BEGIN-BLOCK:
text="begin", type=BEGIN: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="begin_block", type=STATEMENT_BLOCK:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v begin {;;;}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="begin", type=BEGIN:
        text="begin_block", type=STATEMENT_BLOCK:

BLOCKED AST:

BEGIN-BLOCK:
text="begin", type=BEGIN: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="begin_block", type=STATEMENT_BLOCK:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v begin {@x=1}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="begin", type=BEGIN:
        text="begin_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.

BLOCKED AST:

BEGIN-BLOCK:
text="begin", type=BEGIN: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="begin_block", type=STATEMENT_BLOCK:
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="x", type=STRING_LITERAL.
            text="1", type=NUMERIC_LITERAL.

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v begin {@x=1;}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="begin", type=BEGIN:
        text="begin_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.

BLOCKED AST:

BEGIN-BLOCK:
text="begin", type=BEGIN: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="begin_block", type=STATEMENT_BLOCK:
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="x", type=STRING_LITERAL.
            text="1", type=NUMERIC_LITERAL.

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v begin {;@x=1}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="begin", type=BEGIN:
        text="begin_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.

BLOCKED AST:

BEGIN-BLOCK:
text="begin", type=BEGIN: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="begin_block", type=STATEMENT_BLOCK:
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="x", type=STRING_LITERAL.
            text="1", type=NUMERIC_LITERAL.

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v begin {@x=1;@y=2}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="begin", type=BEGIN:
        text="begin_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="y", type=STRING_LITERAL.
                text="2", type=NUMERIC_LITERAL.

BLOCKED AST:

BEGIN-BLOCK:
text="begin", type=BEGIN: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="begin_block", type=STATEMENT_BLOCK:
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="x", type=STRING_LITERAL.
            text="1", type=NUMERIC_LITERAL.
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="y", type=STRING_LITERAL.
            text="2", type=NUMERIC_LITERAL.

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v begin {@x=1;;@y=2}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="begin", type=BEGIN:
        text="begin_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="y", type=STRING_LITERAL.
                text="2", type=NUMERIC_LITERAL.

BLOCKED AST:

BEGIN-BLOCK:
text="begin", type=BEGIN: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="begin_block", type=STATEMENT_BLOCK:
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="x", type=STRING_LITERAL.
            text="1", type=NUMERIC_LITERAL.
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="y", type=STRING_LITERAL.
            text="2", type=NUMERIC_LITERAL.

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v true {}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK:

BLOCKED AST:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=2 max_var_depth=1
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK: subframe_var_count=0


mlr -n put -v true {;}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK:

BLOCKED AST:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=2 max_var_depth=1
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK: subframe_var_count=0


mlr -n put -v true {;;}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK:

BLOCKED AST:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=2 max_var_depth=1
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK: subframe_var_count=0


mlr -n put -v true {;;;}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK:

BLOCKED AST:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=2 max_var_depth=1
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK: subframe_var_count=0


mlr -n put -v true {@x=1}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.

BLOCKED AST:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=2 max_var_depth=1
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK: subframe_var_count=0
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.


mlr -n put -v true {@x=1;}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.

BLOCKED AST:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=2 max_var_depth=1
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK: subframe_var_count=0
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.


mlr -n put -v true {;@x=1}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.

BLOCKED AST:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=2 max_var_depth=1
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK: subframe_var_count=0
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.


mlr -n put -v true {@x=1;@y=2}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="y", type=STRING_LITERAL.
                text="2", type=NUMERIC_LITERAL.

BLOCKED AST:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=2 max_var_depth=1
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK: subframe_var_count=0
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="y", type=STRING_LITERAL.
                text="2", type=NUMERIC_LITERAL.


mlr -n put -v true {@x=1;;@y=2}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="y", type=STRING_LITERAL.
                text="2", type=NUMERIC_LITERAL.

BLOCKED AST:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=2 max_var_depth=1
    text="cond", type=CONDITIONAL_BLOCK:
        text="true", type=BOOLEAN_LITERAL.
        text="cond_block", type=STATEMENT_BLOCK: subframe_var_count=0
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="y", type=STRING_LITERAL.
                text="2", type=NUMERIC_LITERAL.


mlr -n put -v end {}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="end", type=END:
        text="end_block", type=STATEMENT_BLOCK:

BLOCKED AST:

END-BLOCK:
text="end", type=END: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="end_block", type=STATEMENT_BLOCK:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v end {;}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="end", type=END:
        text="end_block", type=STATEMENT_BLOCK:

BLOCKED AST:

END-BLOCK:
text="end", type=END: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="end_block", type=STATEMENT_BLOCK:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v end {;;}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="end", type=END:
        text="end_block", type=STATEMENT_BLOCK:

BLOCKED AST:

END-BLOCK:
text="end", type=END: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="end_block", type=STATEMENT_BLOCK:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v end {;;;}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="end", type=END:
        text="end_block", type=STATEMENT_BLOCK:

BLOCKED AST:

END-BLOCK:
text="end", type=END: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="end_block", type=STATEMENT_BLOCK:

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v end {@x=1}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="end", type=END:
        text="end_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.

BLOCKED AST:

END-BLOCK:
text="end", type=END: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="end_block", type=STATEMENT_BLOCK:
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="x", type=STRING_LITERAL.
            text="1", type=NUMERIC_LITERAL.

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v end {@x=1;}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="end", type=END:
        text="end_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.

BLOCKED AST:

END-BLOCK:
text="end", type=END: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="end_block", type=STATEMENT_BLOCK:
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="x", type=STRING_LITERAL.
            text="1", type=NUMERIC_LITERAL.

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v end {;@x=1}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="end", type=END:
        text="end_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.

BLOCKED AST:

END-BLOCK:
text="end", type=END: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="end_block", type=STATEMENT_BLOCK:
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="x", type=STRING_LITERAL.
            text="1", type=NUMERIC_LITERAL.

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v end {@x=1;@y=2}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="end", type=END:
        text="end_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="y", type=STRING_LITERAL.
                text="2", type=NUMERIC_LITERAL.

BLOCKED AST:

END-BLOCK:
text="end", type=END: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="end_block", type=STATEMENT_BLOCK:
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="x", type=STRING_LITERAL.
            text="1", type=NUMERIC_LITERAL.
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="y", type=STRING_LITERAL.
            text="2", type=NUMERIC_LITERAL.

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


mlr -n put -v end {@x=1;;@y=2}
RAW AST:

AST ROOT:
text="block", type=STATEMENT_BLOCK:
    text="end", type=END:
        text="end_block", type=STATEMENT_BLOCK:
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="x", type=STRING_LITERAL.
                text="1", type=NUMERIC_LITERAL.
            text="=", type=OOSVAR_ASSIGNMENT:
                text="oosvar_keylist", type=OOSVAR_KEYLIST:
                    text="y", type=STRING_LITERAL.
                text="2", type=NUMERIC_LITERAL.

BLOCKED AST:

END-BLOCK:
text="end", type=END: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1
    text="end_block", type=STATEMENT_BLOCK:
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="x", type=STRING_LITERAL.
            text="1", type=NUMERIC_LITERAL.
        text="=", type=OOSVAR_ASSIGNMENT:
            text="oosvar_keylist", type=OOSVAR_KEYLIST:
                text="y", type=STRING_LITERAL.
            text="2", type=NUMERIC_LITERAL.

MAIN BLOCK:
text="main_block", type=STATEMENT_BLOCK: subframe_var_count=1 max_subframe_depth=1 max_var_depth=1


